calico_rkt/calico_rkt.py

```diff
@@ -1,91 +1,249 @@
-import sys
+#!/usr/bin/env python
+from __future__ import print_function
+import socket
+import functools
 import json
-import pycalico
+import os
+import sys
+from subprocess import check_output, CalledProcessError, check_call
+
+from netaddr import IPAddress, IPNetwork, AddrFormatError
+from pycalico import netns
+from pycalico.ipam import IPAMClient, SequentialAssignment
+from pycalico.netns import Namespace
+from pycalico.datastore_datatypes import Rules
+from pycalico.datastore import IF_PREFIX, DatastoreClient
+from pycalico.datastore_errors import PoolNotFound
+
+print_stderr = functools.partial(print, file=sys.stderr)
+
+
+input_ = ''.join(sys.stdin.readlines()).replace('\n', '')
+input_json = json.loads(input_)
+
+# Append to existing env, to avoid losing PATH etc.
+# TODO-PAT: This shouldn't be hardcoded
+env = os.environ.copy()
+ETCD_AUTHORITY_ENV = 'ETCD_AUTHORITY'
+env[ETCD_AUTHORITY_ENV] = 'localhost:2379'
+
+# PROFILE_LABEL = 'CALICO_PROFILE'
+# ETCD_PROFILE_PATH = '/calico/'
+
+ORCHESTRATOR_ID = "rkt"
+HOSTNAME = socket.gethostname()
+NETNS_ROOT= '/var/lib/rkt/pods/run'
 
 def main():
-    stdin_raw_data = sys.stdin.read()
+    mode = env['CNI_COMMAND']
+
+    if mode == 'init':
+        print_stderr('No initialization work to perform')
+    elif mode == 'ADD':
+        print_stderr('Executing Calico pod-creation plugin')
+        create(
+            container_id=env['CNI_CONTAINERID']
+            )
+    elif mode == 'DEL':
+        print_stderr('Executing Calico pod-deletion plugin')
+        delete(
+            container_id=env['CNI_CONTAINERID']
+            )
+
+def create(container_id):
+    """"Handle rkt pod-create event."""
+    print_stderr('Configuring pod %s' % container_id, file=sys.stderr)
+    netns_path='%s/%s/%s' % (NETNS_ROOT, container_id, env['CNI_NETNS'])
+    _datastore_client = IPAMClient()
 
-    # Convert input data to JSON object
     try:
-        stdin_json = json.loads(stdin_raw_data)
-    except ValueError as e:
-        quit_with_error(str(e))
-
-    # Extract command
+        endpoint, ip = _create_calico_endpoint(container_id=container_id,
+                                            netns_path=netns_path,
+                                            client=_datastore_client)
+
+        _create_profile(endpoint=endpoint, 
+                        profile_name=input_json['name'], 
+                        ip=ip,
+                        client=_datastore_client)
+    except CalledProcessError as e:
+        print_stderr('Error code %d creating pod networking: %s\n%s' % (
```

Comment:
Just stick this paren on the next line with the rest of it and align it to the string.